File: src/main/python/UserData.java
#!/usr/bin/env python
""" generated source for module UserData """
# package: nl.insomnia247.nailbiter.eldrinkopubbot.telegram
import org.telegram.telegrambots.meta.api.objects.Update

import org.json.JSONObject

import org.telegram.telegrambots.meta.api.objects.Message

import org.apache.logging.log4j.Logger

import org.apache.logging.log4j.LogManager

# 
#  * @author Alex Leontiev
#  
class UserData(object):
    """ generated source for class UserData """
    _chatId = 0
    _username = None
    _firstName = None
    _lastName = None
    _Log = LogManager.getLogger()

    @overloaded
    def __init__(self):
        """ generated source for method __init__ """

    @__init__.register(object, Update)
    def __init___0(self, update):
        """ generated source for method __init___0 """
        m = None
        if update.hasMessage():
            m = update.getMessage()
        elif update.hasCallbackQuery():
            m = update.getCallbackQuery().getMessage()
        else:
            self._Log.error(update)
        self._Log.info(m)
        self._chatId = m.getChatId()
        self._username = m.getChat().getUserName()
        self._firstName = m.getChat().getFirstName()
        self._lastName = m.getChat().getLastName()

    def getChatId(self):
        """ generated source for method getChatId """
        return self._chatId

    def getUserName(self):
        """ generated source for method getUserName """
        return "@" + self._username

    def getLastName(self):
        """ generated source for method getLastName """
        return self._lastName

    def getFirstName(self):
        """ generated source for method getFirstName """
        return self._firstName

    def __str__(self):
        """ generated source for method toString """
        return Long.toString(self._chatId)

    def toJsonString(self):
        """ generated source for method toJsonString """
        return JSONObject().put("chatId", self._chatId).put("username", self._username).__str__()

File: nl/insomnia247/nailbiter/eldrinkopubbot/telegram/user_data.py
"""===============================================================================

        FILE: nl/insomnia247/nailbiter/eldrinkopubbot/telegram/user_data.py

       USAGE: (not intended to be directly executed)

 DESCRIPTION: 

     OPTIONS: ---
REQUIREMENTS: ---
        BUGS: ---
       NOTES: ---
      AUTHOR: Alex Leontiev (alozz1991@gmail.com)
ORGANIZATION: 
     VERSION: ---
     CREATED: 2021-02-09T21:56:08.940265
    REVISION: ---

==============================================================================="""

class UserData:
    def __init__(self,update):
        self._chat_id = update.effective_message.chat_id
        effective_user = update.effective_user
        self._username = effective_user.username
        self._first_name = effective_user.first_name
        self._last_name = effective_user.last_name
    def __str__(self):
        return str(self._chat_id)
    @property
    def username(self):
        return self._username
    @property
    def first_name(self):
        return self._first_name
    @property
    def last_name(self):
        return self._last_name
